mergify_engine/actions/queue.py

```diff
@@ -0,0 +1,109 @@
+# -*- encoding: utf-8 -*-
+#
+# Copyright © 2020 Mergify SAS
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+import dataclasses
+import typing
+
+import daiquiri
+import voluptuous
+
+from mergify_engine import check_api
+from mergify_engine import context
+from mergify_engine import queue
+from mergify_engine import rules
+from mergify_engine.actions import merge_base
+
+
+LOG = daiquiri.getLogger(__name__)
+
+
+class QueueAction(merge_base.MergeBaseAction):
+    validator = {
+        voluptuous.Required("name"): str,
+        voluptuous.Required(
+            "priority", default=merge_base.PriorityAliases.medium.value
+        ): voluptuous.All(
+            voluptuous.Any("low", "medium", "high", int),
+            voluptuous.Coerce(merge_base.Priority),
+            int,
+            voluptuous.Range(min=1, max=10000),
+        ),
+    }
+
+    queue_rule: rules.QueueRule = dataclasses.field(init=False)
+
+    def dispatch_config(self, mergify_config: typing.Dict):
```

Comment:
```suggestion
    def dispatch_config(self, mergify_config: someCustomType):
```
Might be worth defining a custom type to identify the config, even if it's just an "alias" to `typing.DIct`.

Could be a:
```
MergifyConfig_T = typing.NewType("MergifyConfig", typing.Dict[str, typing.Any])
```
as a starter, until we have a custom class maybe one day?
or just make a class? I don't know.

Reply:
I think the alias is the thing to go for the thing for objects that come from voluptuous